coverage/data.py

```diff
@@ -145,11 +145,17 @@ def combine_parallel_data(
             # we print the original value of f instead of its relative path
             rel_file_name = f
 
-        with open(f, "rb") as fobj:
-            hasher = hashlib.new("sha3_256")
-            hasher.update(fobj.read())
-            sha = hasher.digest()
-            combine_this_one = sha not in file_hashes
+        try:
+            with open(f, "rb") as fobj:
+                hasher = hashlib.new("sha3_256")
+                hasher.update(fobj.read())
+                sha = hasher.digest()
+                combine_this_one = sha not in file_hashes
+        except FileNotFoundError as exc:
+            if data._warn:
+                data._warn(str(exc))
+            if message:
+                message(f"Couldn't combine data file {rel_file_name}: {exc}")
```

Comment:
It's better to keep the try/except as narrow as possible, and don't we have to make sure `combine_this_one` is defined?
```suggestion
        try:
            fobj = open(f, "rb")
        except FileNotFoundError as exc:
            if data._warn:
                data._warn(str(exc))
            if message:
                message(f"Couldn't combine data file {rel_file_name}: {exc}")
            combine_this_one = False
        else:
            with fobj:
                hasher = hashlib.new("sha3_256")
                hasher.update(fobj.read())
                sha = hasher.digest()
                combine_this_one = sha not in file_hashes
```

Reply:
> It's better to keep the try/except as narrow as possible, and don't we have to make sure `combine_this_one` is defined?

Good catch. I agree, and thanks for the patch.